### PR TITLE
Add damask

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

File: recipes/damask/build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# Python Installation 
+cp -r python/damask ${STDLIB_DIR}
+cp VERSION ${PREFIX}/lib/VERSION
+
+# Build Damask
+mkdir build
+cd build 
+cmake -DDAMASK_SOLVER="SPECTRAL" -DCMAKE_INSTALL_PREFIX="${PREFIX}/bin" ..
+make install
```
Comment (chrisburr) [inline recipes/damask/build.sh]:
```suggestion
make -j$CPU_COUNT install
```

File: recipes/damask/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+    - {{ cdt('mesa-dri-drivers') }}  # [linux]
+    - {{ cdt('libselinux') }}  # [linux]
+    - {{ cdt('libxdamage') }}  # [linux]
+    - {{ cdt('libxxf86vm') }}  # [linux] 
+    - make
+    - cmake
+    - hdf5 =*=mpi_mpich_*
```
Comment (chrisburr) [inline recipes/damask/meta.yaml]: This doesn’t belong in build 
```suggestion
```

File: recipes/damask/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    - fftw =*=mpi_mpich_*
```
Comment (chrisburr) [inline recipes/damask/meta.yaml]:
```suggestion
```

File: recipes/damask/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    - vtk
+    - hdf5 =*=mpi_mpich_*
+    - fftw =*=mpi_mpich_*
+    # - h5py
+    - scipy
+  run:
+    - python
+    - petsc =3.10
+    - perl
+    - numpy
```
Comment (chrisburr) [inline recipes/damask/meta.yaml]: Are you using the C API here? If you are you probably need pin_compatible. If not you probably don’t need it in host

Comment (jan-janssen) [reply recipes/damask/meta.yaml]: DAMASK is a Fortran code with python utilities for pre- and post-processing. Only these python utilities depend on numpy but there are not using the C API, just pure Python.

**Pull request merged**